src/Service/MailService.php

```diff
@@ -92,6 +96,9 @@ public function send($email, array $options = []): ResultInterface
         // Trigger pre send event, and cancel email sending if any listener returned false
         $eventResp = $this->dispatcher->dispatch(new PreSendEvent($email));
         if ($eventResp->contains(false)) {
+            if ($this->throwOnCancel) {
+                throw new Exception\MailCancelledException('Email cancelled from pre send event listener');
```

Comment:
As per the comment in which I suggest using a named constructor for the exception, this could be changed to:

```suggestion
                throw Exception\MailCancelledException::fromEvent(PreSendEvent::class);
```